OpenChange/iCalEvent+MAPIStore.m

```diff
@@ -457,7 +460,14 @@ - (void) updateFromMAPIProperties: (NSDictionary *) properties
                     {
                       [person setParticipationStatus: newPartStat];
                       [person setRsvp: @"TRUE"];
-                      [person setRole: @"REQ-PARTICIPANT"];
+                      if (i < ccFirstIndex)
```

Comment:
For me, the real issue here is:
1. Having a hundred lines of code in a for loop
2. This patch now introduces unneeded complexity to read it, i.e. `if (i < ccFirstIndex)`.
    Now, `ccFirstIndex` is defined some 100 lines above and it depends on the logic that we have
    two arrays concatenated.
Imo, the body of the `for` loop should be moved into separate function and then split into to 2 loops.
One for Required participants and one for Optional ones - nice and clear :)
Just my 2c